### PR TITLE
Vb 1921 user information from request header and save on database

### PR DESCRIPTION
**Pull request by dhirajshettymoj**
## What does this pull request do?

Gets the user name information from request header instead of token as the orchestration service will connect to visit-scheduler via client token.

## What is the intent behind these changes?
To reflect changes to the way the orchestration service will connect to visit-scheduler.

### CODE REVIEW

**Comment by rj-adams** [comment]
I'm wondering if we need this extra header:
If the client ID contains the “sub” to the user, then is this sufficient? is it necessary to also pass the username as a separate header?

or do I misunderstand this

**Comment by dhirajshettymoj** [comment]
Code updated to get actionedBy from request body for reserve and cancel visit. Also merged from develop branch.

File: build.gradle.kts [hunk 1]
```diff
@@ -59,6 +59,7 @@ dependencies {
   testImplementation("org.springframework.security:spring-security-test")
   testImplementation("org.testcontainers:postgresql:1.17.6")
   testImplementation("org.testcontainers:localstack:1.17.6")
+  testImplementation("com.amazonaws:aws-java-sdk-s3:1.12.13")
```
Comment (AledWEvans) [inline build.gradle.kts:62]: odd this is, I don't have this on mine and everything works, Has this got something to do with the docker/local host stuff you did?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/controller/VisitControllerLegacy.kt [hunk 2]
```diff
@@ -78,8 +79,8 @@ class VisitControllerLegacy(
     @PathVariable
     reference: String,
     @RequestBody @Valid
-    cancelOutcome: OutcomeDto,
+    cancelVisitDto: CancelVisitDto,
   ): VisitDto {
-    return visitService.cancelVisit(reference.trim(), cancelOutcome)
+    return visitService.cancelVisit(reference.trim(), cancelVisitDto)
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/controller/VisitControllerLegacy.kt:84]: do we want to put it on legacy, as syscon may be still using it, I think they might have started using the new rest end point probably worth checking! I've asked @lbennett-moj  about it!

**Review by AledWEvans — APPROVED** [review]
LGTM  but would check with syscon and perhaps move old way into legacy if no longer using it.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/controller/VisitControllerLegacy.kt (hunk 2 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/controller/VisitControllerLegacy.kt:84]: thanks.. Yes I will speak to syscon

**Pull request merged**